### PR TITLE
update required check for language strings 

### PR DESCRIPTION
**Pull request by Asergey91**
tested with the following form 
```
    ##########################################################
    # Field: Kost description
    ##########################################################
    ext:costDescriptionF a form:Field ;
        sh:name "Beschrijving kost" ;
        form:language "nl";
        sh:order 3 ;
        sh:path dct:description ;
        form:displayType displayTypes:richText ;
        form:validations [
          a form:RequiredConstraint ;
          form:grouping form:Bag ;
          form:language "nl";
          sh:resultMessage "Dit veld is verplicht." ;
          sh:path dct:description ];
        sh:group ext:costsPg-sub .

    ##########################################################
    # Field (En.): Kost description
    ##########################################################
    extEng:costDescriptionF a form:Field ;
        sh:name "Engelse vertaling beschrijving" ;
        sh:order 4 ;
        form:language "en";
        sh:path dct:description ;
        form:displayType displayTypes:richText ;
        form:validations [
          a form:RequiredConstraint ;
          form:grouping form:Bag ;
          form:language "en";
          sh:resultMessage "Dit veld is verplicht." ;
          sh:path dct:description ];
        sh:group ext:costsPg-sub .
```
might have missed an edge case or 2... this is kinda hard to test

### CODE REVIEW

File: constraints/required.js [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
-  return (values.length > 0 && true) || false;
-}
+import { FORM } from '../namespaces';
+
+export default function constraintsRequired(values, options) {
+  if(values.length === 0){
+    return false;
+  }
+  else if(values.length === 1){
+    if(values[0].value?.length === 0){
```
Comment (cecemel) [inline constraints/required.js:8]: I think you still need to check language

File: constraints/required.js [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
+      return false;
+    }
+  }
+  else if(values.length > 1){
```
Comment (cecemel) [inline constraints/required.js:12]: Wouldn't it be easier to check write something along the lines
```
//else (i.e. values.length > 0)  {
  //if the constraint is with language
    // do language check boilerplate
 // else return normal boilerplate
}
```

File: constraints/required.js (hunk 1 above)
Comment (cecemel) [inline constraints/required.js:8]: also: good that you check for empty string (a bit scared though for breaking current forms) 
But could you be consistent and do this in the else branch too then?

File: constraints/required.js [hunk 3]
```diff
@@ -1,3 +1,23 @@
...
+    //for input with multiple languages
+    const language = options.store.match( options.constraintUri, FORM('language'), undefined)[0]?.object?.value;
+    if(language){
+      const value=values.find(value=>value.language==language);
+      if(value?.value?.length === 0){
+        return false
```
Comment (cecemel) [inline constraints/required.js:18]: hence check for empty string here

**Review by cecemel — CHANGES_REQUESTED** [review]
